src/server.c

```diff
@@ -767,9 +778,14 @@ int main(int argc, char *argv[]) {
   soup_server_disconnect(soup_server);
   g_object_unref(soup_server);
 
-  restraint_free_app_data(app_data);
-
   g_main_loop_unref(loop);
 
+  if (!app_data->stdin) {
+      log_manager = rstrnt_log_manager_get_instance ();
```

Comment:
Perhaps create an API which does something to the affect of release_log_manager().  In doing so, I don't think you need the following defined in the logging.h file.   RSTRNT_TYPE_LOG_MANAGER, and G_DECLARE_FINAL_TYPE, and rstrnt_log_manager_get_instance().

Reply:
https://developer.gnome.org/gobject/stable/chapter-gobject.html#gobject-instantiation